src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnCache.java

```diff
@@ -206,7 +203,8 @@ public CacheHandle lookup(Spawn spawn, SpawnExecutionContext context)
 
     context.prefetchInputs();
 
-    if (options.remoteUploadLocalResults) {
+    if (options.remoteUploadLocalResults ||
+        (options.incompatibleRemoteResultsDiskOnly && options.diskCache != null && !options.diskCache.isEmpty())) {
```

Comment:
Can we move the `options.diskCache != null && !options.diskCache.isEmpty())` part to a helper `usesDiskCache` method?

Reply:
Good idea.